lib/hexo/post.js

```diff
@@ -247,7 +247,17 @@ class Post {
     }
 
     // disable Nunjucks when the renderer specify that.
-    const disableNunjucks = ext && ctx.render.renderer.get(ext) && !!ctx.render.renderer.get(ext).disableNunjucks;
+    let disableNunjucks = false;
+    let extRenderer = ext && ctx.render.renderer.get(ext);
+    if (extRenderer) {
+      disableNunjucks = extRenderer.disableNunjucks === true;
+      if (disableNunjucks === false) {
+        extRenderer = ctx.render.renderer.get(ext, true);
+        if (extRenderer) {
+          disableNunjucks = extRenderer.disableNunjucks === true;
+        }
+      }
+    }
```

Comment:
not the best approach, suggestion welcome.

Reply:
Async mode has higher priority in `post.js`. Thus if `async` one has `disableNunjucks` then there is no need to the check `sync` one.